lib/internal/util/inspect.js

```diff
@@ -519,11 +524,11 @@ function formatValue(ctx, value, recurseTimes, typedArray) {
       // This makes sure the recurseTimes are reported as before while using
```

Comment:
☝️ `maybeCustom = value[customInspectSymbol]` uses the unwrapped value to get the `customInspectSymbol` value. However, in usage I've seen folks rely on the `value[customInspectSymbol]` result being that of the `get` trap for `customInspectSymbol`. Would you be up for checking it behind a try-catch?

Reply:
That would contradict the actual intention of this PR: prevent traps from being called similar to browsers. Otherwise we'll keep on getting requests about proxied values not being inspectable.

I'll run CITGM after the security release is done to see if we find any impacted modules.

Reply:
The core of what they're trying to accomplish is provide a customizer without one needing to be bolted on to the value(s) being inspected. Perhaps that could be an inspect option?

Reply:
That would be possible but is really superior to adding the custom inspect symbol? The symbol is easily available (`Symbol.for('nodejs.util.inspect.custom')`) and adding it to the concrete code makes sure everyone is aware of what happens by looking at the code (it's less implicit). Can you post an example where a generic option would be beneficial?

Reply:
The benefit is that you can customize inspection of objects you don't own. Some may not like bolting on symbol properties to objects they don't own because maybe the objects could be frozen, or have their own traps to contend with.

Reply:
Can you provide a code example how this should look like?

Reply:
Something like:
```js
util.inspect(object, {
  customInspect(value, depth, ctx) {
    // do stuff
  }
})
```

Reply:
I meant some code where this would be used in :-)

Because it seems like the option would not work for the outlined use case above.

```js
const foreignObject = {}

const handler = { get() { return 'foo' } }
const proxy = new Proxy(foreignObject, handler)

console.log(proxy)
// Originally: Object [foo] {}
// Now: {}
```

A custom inspect function would not influence this. The above seems somewhat obsolete as it would be the same to directly inspect `do stuff` instead of calling `util.inspect()` with a custom inspect function that in turn returns `do stuff` to inspect.

Reply:
I'm not understanding the ask or the example code below. In my mind it's very clear the distinction. Bolt-on customizers by way of symbols are the object owners way to suggest how an object should be inspected and a customizer option is a way for the person inspecting to suggest how objects should be inspected.

Reply:
> a customizer option is a way for the dev inspecting to suggest how objects should be inspected

I absolutely agree and that is the crux in this case: using a generic custom inspect function as the dev inspecting the object is the same as inspecting the value from the custom inspect function directly. Or do I miss something?

Reply:
For the cases where no customization is needed the customizer could defer to `inspect` feeding it the context/depth from the customizer invocation arguments.

Reply:
AFAIC that's always possible without a custom inspect function.

Reply:
Without a custom function the user would lack `ctx` and `depth`.

Reply:
With a customizer option `ctx` and `depth` would always be indentical to the defaults combined with the passed through options (and some internal state that is going to be removed soon). The reason for that is that it would never come below the main option part as we would not know for what part the customizer should be used.

```js
util.inspect(object, {
  customInspect(value, depth, ctx) {
    // Will always trigger with `value === object` and nothing else.
    // `depth === util.inspect.defaultOptions.depth`
    // `ctx` ==> is in this case identical to `util.inspect.defaultOptions` besides some internal state that is about to be removed.
  }
})
```